unlock-app/src/components/creator/members/GrantKeysDrawer.tsx

```diff
@@ -0,0 +1,316 @@
+            type="datetime-local"
+            {...register('expiration')}
+          />
+          <p className="-mt-4 text-xs italic">
+            This is pre-filled based on the default duration of your lock.
+          </p>
+        </div>
+      </div>
+
+      <div className="flex flex-wrap mb-6 -mx-3">
+        <div className="w-full px-3">
+          <Label htmlFor="grid-key-manager">Key Manager</Label>
+          <Input
+            id="grid-key-manager"
+            type="text"
+            placeholder="0x..."
+            {...register('keyManager', {
+              pattern: ACCOUNT_REGEXP,
+              onChange: addressFieldChanged('keyManager'),
+            })}
+          />
+          {errors.keyManager && (
+            <p className="text-xs -mt-4 text-[#f24c15]">
+              This Ethereum address is not valid.
+            </p>
+          )}
+
+          {!errors.keyManager && (
+            <p className="-mt-4 text-xs italic">
+              If set the key manager has the transfer and cancellation rights
+              for the recipient&apos;s key. If you leave empty, your address
+              will be set as manager.
+            </p>
+          )}
+        </div>
+      </div>
+
+      {!loading && <Button type="submit">Grant Key</Button>}
+      {loading && network && (
+        <TransactionPendingButton network={network} transaction={transaction} />
+      )}
+    </form>
+  )
+}
+
+interface GrantKeysDrawerInterface {
+  isOpen: boolean
+  setIsOpen: (isOpen: boolean) => void
+  lockAddresses: string[]
+}
+
+export const GrantKeysDrawer = ({
+  isOpen,
+  setIsOpen,
+  lockAddresses,
+}: GrantKeysDrawerInterface) => {
+  const { network, account } = useContext(AuthenticationContext) // TODO: use the actual lock network instead of the currently connected network
+  const { openAlert, alertProps } = useAlert()
+
+  const web3Service = useContext(Web3ServiceContext)
+  const [locks, setLocks] = useState<any>({})
+  const [lock, setLock] = useState<any>(null)
+
+  // Let's load the locks's details
+  useEffect(() => {
+    const loadLocks = async (lockAddresses: string[]) => {
+      const locks: Locks = {}
+      await Promise.all(
+        lockAddresses.map(async (address: string) => {
+          locks[address] = await web3Service.getLock(address, network)
+          locks[address].address = address // FIXME getLock does not set address on the lock object...
+
+          // Look if the current user can grant keys
+          locks[address].canGrant = await web3Service.isLockManager(
+            address,
+            account,
+            network
+          )
+          if (!locks[address].canGrant) {
+            locks[address].canGrant = await web3Service.isKeyGranter(
+              address,
+              account,
+              network
+            )
+          }
+        })
+      )
+      setLock(locks[lockAddresses[0]])
+      setLocks(locks)
+    }
+    loadLocks(lockAddresses)
+  }, [lockAddresses.join('')])
```

Comment:
I kind of feel like this could be a separate hook of its own. I think this is reusable but for now, it's okay.